src/commands/vault/channel/processor.ts

```diff
@@ -0,0 +1,74 @@
+import config from "adapters/config"
+import { CommandInteraction, GuildMember } from "discord.js"
+import { GuildIdNotFoundError } from "errors"
+import { MessageEmbed } from "discord.js"
+import { APIError } from "errors"
+import { getEmoji, hasAdministrator, msgColors } from "utils/common"
+import { getErrorEmbed } from "ui/discord/embed"
+
+export async function runCreateChannel({
+  i,
+  guildId,
+}: {
+  i: CommandInteraction
+  guildId?: string | null
+}) {
+  if (!guildId) {
+    throw new GuildIdNotFoundError({ message: i })
+  }
+
+  const member = i.member as GuildMember
+  if (!hasAdministrator(member)) {
+    return {
+      messageOptions: {
+        embeds: [
+          getErrorEmbed({
+            description: "You don't have permission to do this!",
+          }),
+        ],
+      },
+    }
+  }
+
+  const channelArg = i.options.getString("channel")
```

Comment:
discordjs gives us getChannel func, so we don't need to parse channel manually
```suggestion
  const channel = interaction.options.getChannel('destination')
```